pkg/syslog/client.go

```diff
@@ -0,0 +1,44 @@
+package syslog
+
+import (
+	"net"
+
+	"github.com/flynn/flynn/pkg/syslog/rfc5424"
+	"github.com/flynn/flynn/pkg/syslog/rfc6587"
+)
+
+type Client struct {
+	c net.Conn
+
+	h *rfc5424.Header
+	w *rfc5424.Writer
+}
+
+func Dial(address string, hdr *rfc5424.Header) (*Client, error) {
+	conn, err := net.Dial("tcp", address)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		c: conn,
+		h: hdr,
+		w: rfc5424.NewWriter(rfc6587.NewWriter(conn)),
```

Comment:
This seems a little over-abstracted, especially with this 3-level wrapped `Writer`.

Also, it doesn't feel idiomatic for a `func Dial()` to take a struct arg like `hdr` here. Maybe you want a constructor that returns the `*Client`, then a separate method you can call for it to `Dial()` or `Connect()` ?

You can also easily just serialize the `hdr` into `[]byte` once during construction, rather than on every write, since it never changes.